### PR TITLE
tools: Fix `block_search` endpoint integration tests

### LINKED ISSUE #998: `block_search` endpoint isn't tested and doesn't work in CI integration tests
After merging #991, I realized that I didn't see that the [`block_search` test](https://github.com/informalsystems/tendermint-rs/blob/34c5d2a9fde36f9e43a9bfead7e4da5409e14252/tools/kvstore-test/tests/tendermint.rs#L141) isn't actually executed during integration testing, and it actually fails when it's enabled.

**Steps to reproduce**

Add `#[tokio::test]` before the `block_search` method in the kvstore integration test.

**What's the definition of "done" for this issue?**

When all kvstore integration tests pass.

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #998

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Added entry in `.changelog/`

### CODE REVIEW

File: tools/kvstore-test/tests/tendermint.rs [hunk 1]
```diff
@@ -138,18 +138,14 @@ mod rpc {
...
+    #[tokio::test]
     async fn block_search() {
         let res = localhost_http_client()
-            .block_search(
-                Query::gt("block.height", "1"),
-                1,
-                1,
-                Order::Ascending,
-            )
+            .block_search(Query::gt("block.height", 1), 1, 1, Order::Ascending)
```
Comment (thanethomson) [inline tools/kvstore-test/tests/tendermint.rs:145]: The original string representation was breaking CI.

File: tools/kvstore-test/tests/tendermint.rs [hunk 2]
```diff
@@ -264,15 +260,21 @@ mod rpc {
     }
 
     async fn transaction_by_hash() {
+        let rpc_client = localhost_http_client();
```
Comment (thanethomson) [inline tools/kvstore-test/tests/tendermint.rs:263]: These tests kept failing locally so I had to tweak them here too.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -125,7 +125,7 @@ jobs:
     runs-on: ubuntu-latest
     services:
       tendermint:
-        image: informaldev/tendermint:0.34.0
+        image: informaldev/tendermint:0.34.13
```
Comment (thanethomson) [inline .github/workflows/test.yml:128]: Important to note here that the `block_search` endpoint was only introduced in [Tendermint v0.34.9](https://github.com/tendermint/tendermint/blob/master/CHANGELOG.md#v0349).

**Review by xla — APPROVED** [review]
💢 🍆 📹 🍷

**Pull request merged**